Add rule to block creation/editing of reserved namespaces (#8)
appuio/component-appuio-cloud: docs/modules/ROOT/pages/references/parameters.adoc

The namespace in which to deploy this component.

== `reservedNamespaces.match`

[horizontal]
type:: dict

> Creating and editing namespaces matching any of the entries is disallowed.

Reviewer: Hm, the suggestion is not grammatically correct. It's actually fine as it is.

The namespace in which to deploy this component.

> == `reservedNamespaces.allowedClusterRoles.match`

Reviewer: ```suggestion
== `reservedNamespaces.match`
```

[horizontal]
type:: dict

> Creating or editing namespaces matching any of the entries is refuted.

Reviewer: ```suggestion
Creating or editing namespaces matching any of the entries isn't allowed.
```

Must consist of entries like `NAME: {pattern: 'test-*'}`.
Remove entry by setting value to null: `NAME: null`.

== `reservedNamespaces.allowedClusterRoles.match.*.pattern`

[horizontal]
type:: string

Creating or editing namespaces matching any of the entries is refuted.

== `reservedNamespaces.allowedClusterRoles.create`

[horizontal]
type:: boolean
default:: `true`

If enabled creates a `ClusterRole` allowed to create and edit namespaces.

== `reservedNamespaces.allowedClusterRoles.name`

[horizontal]
type:: string
default:: `appuio:edit-reserved-namespaces`

Name of the created `ClusterRole` allowed to create and edit namespaces.

== `reservedNamespaces.allowedClusterRoles.additionalRoles`

[horizontal]
type:: dict
default:: `cluster-admin: {name: 'cluster-admin'}`

`ClusterRoles` exempt from the namespace rules.
Must consist of entries like `NAME: {name: 'my-custom-admin'}`.
Remove entry by setting value to null: `NAME: null`.

== `reservedNamespaces.allowedClusterRoles.additionalRoles.*.name`

[horizontal]
type:: string

> `ClusterRoles` exempt from the namespace rules.

Reviewer: ```suggestion
`ClusterRoles` excluded from the namespace rules.
```
